platform-sdk/swirlds-base/src/main/java/com/swirlds/base/utility/Pair.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.base.utility;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+
+/**
+ * A simple class to hold a pair of values.
+ *
+ * @param left value for the left/key side of the pair
+ * @param right value for the right/value side of the pair
+ * @param <L> type of the left/key side of the pair
+ * @param <R> type of the right/value side of the pair
+ */
+public record Pair<L, R>(L left, R right) {
+    /**
+     * Create a new pair of values.
+     *
+     * @param left value for the left/key side of the pair
+     * @param right value for the right/value side of the pair
+     * @param <L> type of the left/key side of the pair
+     * @param <R> type of the right/value side of the pair
+     *
+     * @return a new pair of key and value
+     */
+    @NonNull
+    public static <L, R> Pair<L, R> of(@Nullable final L left, @Nullable final R right) {
```

Comment:
We should create an issue that left / right is renamed to key/value. In addition we should check if we can define key as NonNull.